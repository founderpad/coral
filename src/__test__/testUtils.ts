// import { ThemeProvider } from "my-ui-lib"
// import { TranslationProvider } from "my-i18n-lib"
// import defaultStrings from "i18n/en-x-default"
import { render } from '@testing-library/react';
import { JSXElementConstructor, ReactElement } from 'react';

const Providers = ({ children }: { children: any }) => {
	return children;
	// return (
	//   <ThemeProvider theme="light">
	//     <TranslationProvider messages={defaultStrings}>
	//       {children}
	//     </TranslationProvider>
	//   </ThemeProvider>
	// )
};

const customRender = (
	ui: ReactElement<any, string | JSXElementConstructor<any>>,
	options = {}
): any => render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
