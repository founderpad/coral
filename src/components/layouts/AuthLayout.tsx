import { Heading } from '@chakra-ui/layout';
import { Tag } from '@chakra-ui/tag';
import { DocumentTitle } from '@components/shared';
import FounderpadLogo from '@components/shared/FounderpadLogo';
import React from 'react';
import { FlexLayout } from './FlexLayout';
import { StackLayout } from './StackLayout';

interface Props {
	header: string;
	children: React.ReactNode;
	title: string;
	subheader?: string;
}

const AuthLayout = ({ header, children, title }: Props) => (
	<React.Fragment>
		<DocumentTitle title={title} />
		<StackLayout
			minW={{ base: 'full', sm: '400px' }}
			maxW={{ base: 'full', sm: '450px' }}
			borderColor={'gray.100'}
			p={8}
			boxShadow={{ sm: 'sm' }}
			borderWidth={{ sm: '1px' }}
			// m={'auto'}
			style={{ margin: 'auto' }}
			bg={'white'}
			d={'flex'}
			flex={{ base: 1, sm: 'none' }}
			justifyContent={'center'}
			id={'auth-container'}
		>
			<FlexLayout mx={'auto'} alignItems={'center'}>
				<FounderpadLogo />
				<Tag
					bg={'fpPrimary.700'}
					color={'white'}
					textTransform={'capitalize'}
					fontWeight={'medium'}
					textAlign={'center'}
					verticalAlign={'center'}
					fontSize={'sm'}
					rounded={'sm'}
				>
					Beta
				</Tag>
			</FlexLayout>
			<Heading
				textAlign={'center'}
				fontWeight={'normal'}
				fontSize={'md'}
				color={'black'}
				py={4}
			>
				{header}
			</Heading>
			{children}
		</StackLayout>
	</React.Fragment>
);

export default AuthLayout;
