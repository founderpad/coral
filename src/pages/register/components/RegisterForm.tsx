import { SubmitButton } from '@components/buttons';
import { Form } from '@components/form';
import {
	EmailField,
	InputField,
	PasswordField
} from '@components/input/InputField';
// import SocialLogins from '@components/shared/SocialLogins';
import { useRegister } from '@hooks/auth';
import React from 'react';
import { useForm } from 'react-hook-form';
import { IRegisterFormData } from 'src/types/auth';
import LegalFooter from './LegalFooter';

const RegisterForm = () => {
	const {
		handleSubmit,
		control,
		formState: { errors, isSubmitting, isValid }
	} = useForm<IRegisterFormData>({ mode: 'all' });
	const onRegister = useRegister();

	return (
		<React.Fragment>
			<Form
				id={'registerForm'}
				name={'registerForm'}
				onSubmit={handleSubmit(onRegister)}
				stackProps={{
					alignItems: 'center'
				}}
			>
				<InputField
					name="firstName"
					error={errors.firstName}
					errorText="You must input a first name"
					placeholder="First name"
					fontSize={'sm'}
					control={control}
					rules={{ maxLength: 20, minLength: 2 }}
					size={'md'}
					isRequired
				/>
				<InputField
					name="lastName"
					placeholder="Last name"
					control={control}
					rules={{ minLength: 0, maxLength: 20 }}
					size={'md'}
					fontSize={'sm'}
				/>
				<EmailField
					id="email"
					name="email"
					error={errors['email']}
					control={control}
					size={'md'}
					fontSize={'sm'}
					isRequired
				/>
				<PasswordField
					id="password"
					name="password"
					error={errors['password']}
					rules={{ maxLength: 20, minLength: 6 }}
					control={control}
					size={'md'}
					fontSize={'sm'}
					isRequired
				/>
				{/* <AlertFeedback /> */}
				<SubmitButton
					id={'submit-register-account'}
					name={'submit-register-account'}
					label={'Create account'}
					isLoading={isSubmitting}
					disabled={!isValid || isSubmitting}
					// disabled={isSubmitting}
					size={'md'}
					fontSize={'sm'}
					w={{ base: 'full', sm: '175px' }}
				/>
			</Form>
			{/* <SocialLogins /> */}
			<LegalFooter />
		</React.Fragment>
	);
};

export default RegisterForm;
