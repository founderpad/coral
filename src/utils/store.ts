import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { persistReducer } from 'redux-persist';
import { PersistConfig } from 'redux-persist/es/types';
import storage from 'redux-persist/lib/storage';
import rootReducer from './reducer';

/**
 * Create the Redux store
 *
 * @author jlee
 */

// const isProd = process.env.NODE_ENV !== 'development';

/**
 * Persist storage configuration set up
 * @key[root]
 * @storage[storage]
 */
const persistConfig: PersistConfig<any> = {
	key: 'root',
	storage
};

const middleware = [...getDefaultMiddleware({ serializableCheck: false })];

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
	reducer: persistedReducer,
	middleware
	// devTools: isProd ?? false
});

// export { StoreProvider }

export default store;
