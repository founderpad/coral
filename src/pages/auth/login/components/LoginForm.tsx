import { AlertFeedback } from '@components/alert';
import { SubmitButton } from '@components/buttons';
import { BaseForm } from '@components/form';
import { FormInput } from '@components/form/inputs/FormField';
import { Label } from '@components/labels';
import { FlexLayout } from '@components/layouts';
import { PrimaryLink } from '@components/links';
import { useLogin } from '@hooks/auth';
import { useQueryParam } from '@hooks/util';
import { emailPattern } from '@utils/validators';
import React, { memo } from 'react';
import { TLoginFields } from '../../../../types/auth';

const defaultValues = {
	email: '',
	password: ''
};

const LoginForm = () => {
	const onLogin = useLogin();
	const isError = useQueryParam('error');
	const isVerifiedEmail = useQueryParam('type') === 'emailVerify';

	return (
		<React.Fragment>
			{isVerifiedEmail && (
				<AlertFeedback message="Email address successfully verified" />
			)}

			<BaseForm<TLoginFields>
				name="login-form"
				onSubmit={onLogin}
				defaultValues={defaultValues}
				stackProps={{
					alignItems: 'center',
					spacing: 2
				}}
			>
				{({
					register,
					control,
					resetField,
					formState: { errors, isSubmitting }
				}) => (
					<React.Fragment>
						<FormInput<TLoginFields>
							id="email"
							name="email"
							placeholder="Email"
							register={register}
							control={control}
							fieldProps={{
								placeholder: 'Email'
							}}
							rules={{
								required:
									'You must enter a valid email address',
								pattern: emailPattern
							}}
							hideLimit={true}
							errors={errors}
							onClear={() => resetField('email')}
						/>

						<FormInput<TLoginFields>
							id="password"
							name="password"
							placeholder="Password"
							register={register}
							control={control}
							fieldProps={{
								placeholder: 'Password',
								type: 'password'
							}}
							rules={{
								required: 'You must enter a valid password',
								minLength: {
									value: 6,
									message:
										'Your password must be a minimum of 6 characters'
								},
								maxLength: {
									value: 20,
									message:
										'Your password must be a maximum of 20 characters'
								}
							}}
							hideLimit={true}
							errors={errors}
							onClear={() => resetField('password')}
						/>

						{isError && (
							<AlertFeedback
								status="error"
								message={
									'Failed to login. Incorrect email and/or password.'
								}
							/>
						)}

						<SubmitButton
							id="submit-login"
							name="submit-login"
							label="Log in"
							isLoading={isSubmitting}
							disabled={isSubmitting}
							size="md"
							fontSize="small"
							w={{ base: 'full', sm: '150px' }}
						/>
					</React.Fragment>
				)}
			</BaseForm>
			{/* <SocialLogins /> */}
			<LoginFooter />
		</React.Fragment>
	);
};

const LoginFooter = memo(() => (
	<FlexLayout justifyContent="space-between" alignItems="center" pt={8}>
		<Label color="gray.500" fontSize="x-small" alignSelf="center">
			No account?{' '}
			<PrimaryLink href="/register" title="Link to register an account">
				Register now
			</PrimaryLink>
		</Label>
		<PrimaryLink
			href="/resetpassword"
			title="Link to register an account"
			fontSize="x-small"
		>
			Forgotten password?
		</PrimaryLink>
	</FlexLayout>
));

export default LoginForm;
