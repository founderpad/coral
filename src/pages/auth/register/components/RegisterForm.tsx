import { FormInput } from '@/components/form/inputs/FormField';
import { useRegister } from '@/hooks/auth';
import React, { useContext } from 'react';
import LegalFooter from './LegalFooter';
import { TRegisterFormFields } from '../../../../types/auth';
import { BaseForm } from '@/components/form';
import { SubmitButton } from '@/components/buttons';
import { AlertFeedback } from '@/components/alert';
import SocialLogins from '@/components/shared/SocialLogins';
import NotificationContext from '@/context/NotificationContext';
import { schema } from '../validationSchema';

const defaultValues: Record<string, string> & TRegisterFormFields = {
	firstName: '',
	lastName: '',
	email: '',
	password: ''
};

const RegisterForm = () => {
	const { onRegister } = useRegister();
	const { notification } = useContext(NotificationContext);

	return (
		<React.Fragment>
			<BaseForm<TRegisterFormFields, typeof schema>
				name="register-form"
				onSubmit={onRegister}
				defaultValues={defaultValues}
				schema={schema}
				stackProps={{
					alignItems: 'center',
					spacing: 3
				}}
			>
				{({
					register,
					control,
					formState: { errors, isSubmitting }
				}) => (
					<React.Fragment>
						<FormInput<TRegisterFormFields>
							id="firstName"
							name="firstName"
							placeholder="First name"
							register={register}
							control={control}
							errors={errors}
							fieldProps={{
								placeholder: 'First name'
							}}
							hideLimit={true}
							hideClear
						/>

						<FormInput<TRegisterFormFields>
							id="lastName"
							name="lastName"
							placeholder="Last name"
							register={register}
							control={control}
							fieldProps={{
								placeholder: 'Last name'
							}}
							hideLimit={true}
							errors={errors}
							hideClear
						/>
						<FormInput<TRegisterFormFields>
							id="email"
							name="email"
							placeholder="Email"
							register={register}
							control={control}
							fieldProps={{
								placeholder: 'Email'
							}}
							hideLimit={true}
							errors={errors}
							hideClear
						/>

						<FormInput<TRegisterFormFields>
							id="password"
							name="password"
							placeholder="Password"
							register={register}
							control={control}
							fieldProps={{
								placeholder: 'Password',
								type: 'password'
							}}
							hideLimit={true}
							errors={errors}
							hideClear
						/>

						{notification && (
							<AlertFeedback
								status={notification.status}
								message={notification.message}
							/>
						)}

						<SubmitButton
							id="submit-register-account"
							name="submit-register-account"
							label="Create account"
							isLoading={isSubmitting && !notification}
							disabled={isSubmitting && !notification}
							size="md"
							fontSize="small"
							w={{ base: 'full', sm: '200px' }}
						/>
					</React.Fragment>
				)}
			</BaseForm>
			<SocialLogins />

			{/* <Alert fontSize="xs" status="warning" p={2} textAlign="center">
				Your email address must be the same as your PayPal account in
				order to withdraw funds.
			</Alert> */}
			<LegalFooter />
		</React.Fragment>
	);
};

export default RegisterForm;
