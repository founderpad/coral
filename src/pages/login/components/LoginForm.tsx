import { Flex, Text } from '@chakra-ui/layout';
import { SubmitButton } from 'components/buttons';
import { Form } from 'components/form';
import { useBaseForm } from 'components/form/hooks';
import { EmailField, PasswordField } from 'components/input';
import { StackLayout } from 'components/layouts';
import { PrimaryLink } from 'components/links';
import { useLogin } from 'hooks/auth';
import React, { memo } from 'react';
import { IAuthFormData } from 'types/auth';

const LoginForm = (): JSX.Element => {
	const {
		handleSubmit,
		control,
		formState: { errors, isSubmitting, isValid }
	} = useBaseForm<IAuthFormData>();
	const onLogin = useLogin();

	return (
		<Form
			id={'login-form'}
			name={'loginform'}
			onSubmit={handleSubmit(onLogin)}
		>
			<StackLayout spacing={6} my={6}>
				<EmailField
					id="email"
					name="email"
					error={errors['email']}
					errorText="Please enter a valid email"
					control={control}
					isRequired
				/>
				<PasswordField
					id="password"
					name="password"
					error={errors['password']}
					rules={{ maxLength: 20, minLength: 6 }}
					control={control}
					isRequired
				/>

				<SubmitButton
					id={'submit-login'}
					name={'submit-login'}
					label="Login now"
					isLoading={isSubmitting}
					disabled={!isValid || isSubmitting}
					size={'md'}
					full
				/>
				<NoAccountFooter />
			</StackLayout>
		</Form>
	);
};

const NoAccountFooter = memo(
	(): JSX.Element => (
		<Flex
			spacing={6}
			justifyContent={'center'}
			w="full"
			mt={'auto'}
			mx={'auto'}
		>
			<Text color={'gray.500'} fontSize={'xs'}>
				Don&apos;t have an account?
			</Text>
			&nbsp;
			<PrimaryLink
				href="/register"
				fontSize={'xs'}
				title={'Link to register an account'}
			>
				Create one here
			</PrimaryLink>
		</Flex>
	)
);

export default LoginForm;
