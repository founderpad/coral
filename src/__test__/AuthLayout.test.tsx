import LoginForm from 'pages/login/components/LoginForm';
import RegisterForm from 'pages/register/components/RegisterForm';
import { render } from '__test__/testUtils';
import store from 'utils/store';
import { Provider } from 'react-redux';
import AuthLayout from '../components/layouts/AuthLayout';

describe('AuthLayout layout', () => {
	it('matches AuthLayout snapshot', () => {
		const { asFragment } = render(
			<Provider store={store}>
				<AuthLayout
					header="Login"
					subheader="Pick up where you left off"
					title="Login"
				>
					<LoginForm />
				</AuthLayout>
			</Provider>
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('renders AuthLayout with LoginForm form', () => {
		const { getByText } = render(
			<Provider store={store}>
				<AuthLayout
					header="Login"
					subheader="Pick up where you left off"
					title="Login"
				>
					<LoginForm />
				</AuthLayout>
			</Provider>
		);
		expect(getByText(/Pick up where you left off/i)).toBeInTheDocument();
	});

	it('renders AuthLayout with RegisterForm form', () => {
		const { getByText } = render(
			<Provider store={store}>
				<AuthLayout
					header="Register"
					subheader="Launch your vision"
					title="Login"
				>
					<RegisterForm />
				</AuthLayout>
			</Provider>
		);
		expect(getByText('Register')).toBeInTheDocument();
		expect(getByText(/Launch your vision/i)).toBeInTheDocument();
	});
});

// it('clicking button triggers alert', () => {
//     const { getByText } = render(<Home />, {});
//     window.alert = jest.fn();
//     fireEvent.click(getByText('Test Button'));
//     expect(window.alert).toHaveBeenCalledWith('With typescript and Jest');
// });
