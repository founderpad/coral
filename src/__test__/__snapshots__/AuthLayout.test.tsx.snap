// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLayout layout matches AuthLayout snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-11we7xs"
  >
    <img
      alt="logo"
      class="chakra-image css-1b7p44u"
      src="/logo.svg"
    />
    <h2
      class="chakra-heading css-1zp44w"
    >
      Login
    </h2>
    <form
      aria-label="loginform"
      autocomplete="off"
      id="login-form"
      name="loginform"
      novalidate=""
    >
      <div
        class="chakra-stack css-6epcyi"
      >
        <div
          class="chakra-stack css-1moek9b"
        >
          <div
            class="chakra-form-control css-1naxzcr"
            role="group"
          >
            <input
              aria-label="email"
              aria-required="true"
              class="chakra-input css-0"
              control="[object Object]"
              errortext="Please enter a valid email"
              id="email-email"
              name="email"
              placeholder="Email"
              required=""
              rules="[object Object]"
              type="email"
              value=""
            />
          </div>
          <div
            class="chakra-form-control css-1naxzcr"
            role="group"
          >
            <input
              aria-label="password"
              aria-required="true"
              class="chakra-input css-0"
              control="[object Object]"
              errortext="Please enter a valid password between 6 and 20 characters."
              id="password-password"
              name="password"
              placeholder="Password"
              required=""
              rules="[object Object]"
              type="password"
              value=""
            />
          </div>
          <button
            aria-label="submit-login"
            class="chakra-button css-uxt1e8"
            disabled=""
            id="submit-login"
            type="submit"
          >
            Login now
          </button>
          <p
            class="chakra-text css-1izd6ip"
          >
            Don't have an account?
            <a
              class="chakra-link css-qgrdgl"
              href="/app/register"
              title="Link to register an account"
            >
               Create one here
            </a>
          </p>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
